timApp/static/scripts/tim/timRoot.ts

```diff
@@ -212,6 +219,41 @@ export class RootCtrl {
                 }
             }
         }
+        this.checkTosAcceptance();
+    }
+
+    private checkTosAcceptance() {
+        const latestTosDate = documentglobals().latest_tos_date;
+        const tosAcceptedAt = documentglobals().current_user.tos_accepted_at;
+        const currentPage: IDocument | undefined = documentglobals().curr_item;
+        const tosPath = documentglobals().footerDocs.termsOfService;
+        let inTosPage = false;
+
+        if (tosPath && currentPage) {
+            const splitPath = currentPage.path.split("/");
+            const splitFooterPath = tosPath.split("/");
+            // Remove language code if present
+            if (
+                splitPath[splitPath.length - 1] !==
+                splitFooterPath[splitFooterPath.length - 1]
+            ) {
+                splitPath.pop();
+            }
+            const currentPath = splitPath.join("/");
+            inTosPage = currentPath === tosPath;
+        }
```

Comment:
Hmm, minusta koko polun splittaus on ehkä vähän overkill jos vaan halutaan normalisoida pois kielikoodi. Ehdotan ehkä yksinkertaisempaa substringia, koska JS:llä ei ole parametrillista trimEndia


```suggestion
        if (tosPath && currentPage) {
            const langPostfix = currentPage.lang_id
                ? `/${currentPage.lang_id}`
                : "";
            const currentPathNoLangCode = currentPage.path.substring(
                0,
                currentPage.path.length - langPostfix.length
            );
            inTosPage = currentPathNoLangCode === tosPath;
        }
```